app/entity-repositories/pasteRepository.js

```diff
@@ -0,0 +1,106 @@
+const PasteFactory = require('../entities/paste-entity/pasteFactory');
+
+class PasteRepository {
+    constructor ({ dbProvider }) {
+        this.dbProvider = dbProvider;
+        this.pasteFactory = new PasteFactory();
+    }
+
+    async findByID({ id }) {
+
+        const result = await this.findOne({ id });
+        return result;
+
+    }
+
+    async findOne({ id, name, authorID }) {
+        
+        const result = await this.findAll({ ids: id, name, authorID });
+        
+        return result.length > 0 ? result[0] : null;
+
+    }
+
+    async findAll({ ids, name, authorID}) {
+
+        const itemsToFind = [];
+
+        if ( ids ) {
+
+            const arrayOfIDs = [];
+
+            if (ids.length > 1) {
+                
+                for (let i = 0; i < ids.length; i++) {
+                    arrayOfIDs.push(ids[i]);
+                }
+
+                itemsToFind.push(`id in (${arrayOfIDs.join(', ')})`);
```

Comment:
What is going on in your opinion?
You've got `ids: [id1, id2, id3]`
then you map it into `arrayOfIDs: [id1, id2, id3]`
What's a point?)

Reply:
just to be sure